doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
 
-.. note:: TODO. This is a placeholder section; it is not written yet.
+The context for the following examples will be to import the :class:`Graph` class and to have one or more graphs available:
 
+>>> import igraph as ig
+>>> from igraph import Graph
+>>> g = Graph(edges=[[0, 1], [2, 3]])
+
+To get a summary representation of the graph, use :meth:`Graph.summary`. For instance
+
+>>> g.summary(verbosity=1)
+
+will provide a fairly detailed description.
+
+To copy a graph, use :meth:`Graph.copy`.
+
+Vertices and edges
++++++++++++++++++++++++++++
+Vertices are numbered 0 to `n-1`, where n is the number of vertices in the graph. These are called the "vertex ids".
+To count vertices, use :meth:`Graph.vcount`:
+
+>>> n = g.vcount()
+
+Edges also have ids from 0 to `m-1` and are counted by :meth:`Graph.ecount`:
+
+>>> m = g.ecount()
+
+To get a sequence of vertices, use their ids and :attr:`Graph.vs`:
+
+>>> for v in g.vs:
+>>>     print(v)
+
+Similarly for edges, use :attr:`Graph.es`:
+
+>>> for e in g.es:
+>>>     print(e)
+
+.. note:: These two attributes are special sequences with their own useful methods. Some are described below, and all can be found in the `API documentation`_.
+
+If you prefer a vanilla edge list, you can use :meth:`Graph.get_edge_list`.
+
+Incidence
+++++++++++++++++++++++++++++++
+To get the vertices at the two ends of an edge, use :attr:`Edge.source` and :attr:`Edge.target`:
+
+>>> e = g.es[0]
+>>> v1, v2 = e.source, e.target
+
+Vice versa, to get the edge if from the source and target vertices, you can use :meth:`Graph.get_eid` or, for multiple pairs of source/targets,
+:meth:`Graph.get_eids`. The boolean version, asking whether two vertices are directly connected, is :meth:`Graph.are_connected`.
+
+To get the edges incident on a vertex, you can use :meth:`Vertex.incident`, :meth:`Vertex.out_edges` and
+:meth:`Vertex.in_edges`. The three are equivalent on undirected graphs but not directed ones, of course:
+
+>>> v = g.vs[0]
+>>> edges = v.incident()
+
+The :meth:`Graph.adjecent` and :meth:`Graph.incident` functions fulfill the same purpose with a slightly different syntax based on vertex ids:
+
+>>> edges = g.incident(0)
+
+To get the full adjacency/incidence list representation of the graph, use :meth:`Graph.get_adjlist`, :meth:`Graph.g.get_inclist()` or, for a bipartite graph, :meth:`Graph.get_incidence`.
+
+Neighborhood
++++++++++++++
+To compute the neighbors, successors, and predecessors, the methods :meth:`Graph.neighbors`, :meth:`Graph.successors` and
+:meth:`Graph.predecessors` are available. The three give the same answer in undirected graphs and have a similar dual syntax:
+
+>>> neis = g.vs[0].neighbors()
+>>> neis = g.neighbors(0)
+
+To get the list of vertives within a certain distance of one or more initial nodes, you can use :meth:`Graph.neighborhood`:
```

Comment:
vertives -> vertices

"certain distance *from*" (not of)

Since the sentence started with "vertex", let's not switch midway to "node"

Reply:
@ntamas On a related note, it would be useful to standardize on a single naming *for the API* (not the documentation). Either use "vertex" consistently (my preference) or use "node" consistently. Do not use both. 

I would not change existing things, but it would be nice to be consistent going forward.

Reply:
I missed the *within*. Using *of* is fine here.

Reply:
"Vertex" is my preference as well, although another guideline is to use the same name in the Python interface as in the C core. So, for instance, that's why we have `node_compat_fn` and not `vertex_compat_fn`.

Reply:
Alright, I think `from` reads better. I also switched to `vertices`.

Agreed let's try to prefer `vertex/ices` over `node/s` in the API.